### PR TITLE
Feature/n2 158 create csv for videos

### PR DESCRIPTION
**Pull request by Dimithri**
JIRA Ticket: https://ackama.atlassian.net/browse/N2-158

This creates a download for the video file details which are public as a csv. Link to download the file has been added as a link after the gloss csv download link in manage lexicon page.

image

### CODE REVIEW

File: signbank/dictionary/adminviews.py [hunk 1]
```diff
@@ -713,6 +713,7 @@ def get_video_details(request):
     writer = csv.writer(response)
 
     csv_queryset =GlossVideo.objects.filter(is_public=True)
+    # return render_to_csv_response(csv_queryset)
```
Comment (jonholdsworth) [inline signbank/dictionary/adminviews.py]: Are you deliberately leaving this comment?

**Review by jonholdsworth — APPROVED** [review]
Nice work!

File: signbank/dictionary/adminviews.py (hunk 1 above)
Comment (Dimithri) [reply signbank/dictionary/adminviews.py]: No, I'm not. Must have slipped out when I was removing other comments. Will remove it :smile:

File: signbank/dictionary/adminviews.py [hunk 2]
```diff
@@ -698,6 +698,48 @@ def serialize_glosses_csv(dataset, queryset):
...
+
+    for video in csv_queryset:
+        row = list()
+        # Add data from each field.
+        for f in fields:
+            value = str(getattr(video, f.name))
+
+            # If the value contains ';', put it in quotes.
+            if value and ";" in value:
+                row.append('"{}"'.format(value))
```
Comment (joshmcarthur) [inline signbank/dictionary/adminviews.py]: Does Python not handle escaping characters for you?

**Review by joshmcarthur — CHANGES_REQUESTED** [review]
I think that there's a vital piece of information here - the video URL 😆

**Comment by joshmcarthur** [comment]
@Dimithri, @jonholdsworth I'll leave this up to you to review, but I've extended the glossvideo model to support manging the access-control-level of the object in S3 when the glossvideo record is published or unpublished. When the record is published, it will be marked as ACL `public-read`, meaning that no signed URL is needed to access the object. When the record is unpublished, it is marked as `private`, meaning that a signed URL is needed for access. 

This means we can export non-signed URLs in the CSV export. There isn't a way to do this with boto3, but because we know the URL format, we can just generate a string that we know is right. This means the export is super fast, under a second for the entire set of migrated assets.

Finally, I moved the export itself to the video django app, since that's where it should be. It is invoked from the same place, it just uses the video views.py definition.

**Review by jonholdsworth — APPROVED** [review]
I don't know enough to comment on the fine implementation details, but I understand what this change is doing. I couldn't find any obvious coding mistakes and all the dead comments have been cleaned up.
All in all this is great little piece of work folks, well done.

File: signbank/dictionary/adminviews.py (hunk 2 above)
Comment (jonholdsworth) [reply signbank/dictionary/adminviews.py]: Sorry to butt in, but - is it safe to escape characters in a CSV formatted file? Will they not appear with the `'\' `or whatever in the target output?  Or does that not matter in this case?

**Review by joshmcarthur — APPROVED** [review]

**Pull request merged**